src/main/java/com/marklogic/kafka/connect/source/CsvRowBatcherBuilder.java

```diff
@@ -0,0 +1,49 @@
+package com.marklogic.kafka.connect.source;
+
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.RowBatchSuccessListener;
+import com.marklogic.client.datamovement.RowBatcher;
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.io.marker.ContentHandle;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.List;
+import java.util.Map;
+
+public class CsvRowBatcherBuilder extends LoggingObject implements RowBatcherBuilder<String> {
+    private final String topic;
+    private final DataMovementManager dataMovementManager;
+
+    CsvRowBatcherBuilder(DataMovementManager dataMovementManager, Map<String, Object> parsedConfig) {
+        this.dataMovementManager = dataMovementManager;
+        this.topic = (String) parsedConfig.get(MarkLogicSourceConfig.TOPIC);
+    }
+
+    public RowBatcher<String> newRowBatcher(List<SourceRecord> newSourceRecords) {
+        ContentHandle<String> contentHandle = new StringHandle().withFormat(Format.TEXT).withMimetype("text/csv");
+        RowBatcher<String> rowBatcher =  dataMovementManager.newRowBatcher(contentHandle);
+        rowBatcher.onSuccess(event -> onSuccessHandler(event, newSourceRecords));
+        return rowBatcher;
+    }
+
+    protected void onSuccessHandler(RowBatchSuccessListener.RowBatchResponseEvent<String> event, List<SourceRecord> newSourceRecords) {
+        String document = event.getRowsDoc();
+        logger.debug("CSV document: \n" + document);
+        BufferedReader reader = new BufferedReader(new StringReader(document));
+        try {
+            String headers = reader.readLine();
+            reader.lines().forEach(line -> {
+                String newDocument = headers + "\n" + line;
+                SourceRecord newRecord = new SourceRecord(null, null, topic, null, newDocument);
+                newSourceRecords.add(newRecord);
+            });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Java IO is annoying with its checked exceptions - we certainly don't expect this IOException here, but we need to catch it and log it just like in the JSON builder. I think a convenience method would be good here - I would add AbstractRowBatchBuilder as a parent class for each RBB impl and have a "logBatchError(Exception ex)" method that logs the message that the JSON builder is logging.